src/client/scripts/cardModal.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+/* global Cards:false SnackbarMethods:false UserMethods:false */
+
+Template.cardModal.helpers({
+    cardModalInfo: function(_id) {
+        return Cards.findOne({"_id": _id});
+    }
+});
+
+Template.cardModal.events({
+    "click #submitCommentButton": function(eve){
+        eve.preventDefault();
+        var comment = eve.target.form[0].value;
+        var image = null;
+
+        if(!comment || comment.length <= 4)
+            SnackbarMethods.DisplayMessage("Enter a more valuable comment",3000);//eslint-disable-line
+        else{
+            var author;
+
+            if(!Meteor.user())
+                author = Session.get("author");
+            else{
+                author = Meteor.user().profile.name;
+                image = UserMethods.getUserImage(Meteor.user()._id);
+            }
+            var commentToAdd = new Comment().createdBy(author)
+              .withText(comment).createdAtTime(new Date()).withProfile(image);
+
+            Meteor.call("submitComment",this._id,commentToAdd);
+            eve.target.form[0].value = "";
+            $("ul.collapsible li").show();
+            $("i.fa-caret-right").addClass("fa-caret-down");
+            $("i.fa-caret-right").removeClass("fa-caret-right");
+        }
+    },
+
+    "submit #submitCommentButton": function(eve){
```

Comment:
That literally makes no sense...
Delete the click one, keep the submit one. Make sure the button is part of the form and is indicated as the submit of the form. You just then only need to press enter and it will be smart and submit, same with the button.
There is ZERO reason this needs to be done twice.